circuit_knitting/cutting/cutting_experiments.py

```diff
@@ -389,3 +371,80 @@ def _get_pauli_indices(cog: CommutingObservableGroup) -> list[int]:
     if not pauli_indices:
         pauli_indices = [0]
     return pauli_indices
+
+
+def _consolidate_resets(circuit: QuantumCircuit, inplace=True) -> QuantumCircuit:
+    """Consolidate redundant resets into a single reset."""
+    if not inplace:  # pragma: no cover
+        circuit = deepcopy(circuit)
+
+    # Keep up with whether the previous instruction on a given qubit was a reset
+    resets = {i: False for i in range(circuit.num_qubits)}
```

Comment:
should potentially be a bit faster
```suggestion
    resets = [False for i in range(circuit.num_qubits)]
```